tests/sentry/backup/test_imports.py

```diff
@@ -868,38 +860,41 @@ def test_global_import_scoping(self):
             == RegionImportChunk.objects.values("import_uuid").first()
         )
 
-    @mock.patch("sentry.backup.imports.is_split_db", return_value=False)
-    @mock.patch("sentry.backup.imports.reset_models")
-    def test_global_import_initial_model_deletion(self, is_split_db, reset_models):
-        if SiloMode.get_current_mode() != SiloMode.MONOLITH:
-            return
 
-        create_default_projects()
+@region_silo_test
+class DatabaseResetTests(ImportTestCase):
+    """
+    Ensure that database resets work as intended in different import scopes.
+    """
+
+    def import_empty_backup_file(self, import_fn):
         with tempfile.TemporaryDirectory() as tmp_dir:
             tmp_empty_file_path = tmp_dir + "empty_backup.json"
             with open(tmp_empty_file_path, "w") as tmp_file:
                 json.dump([], tmp_file)
             with open(tmp_empty_file_path, "rb") as empty_backup_json:
-                with assume_test_silo_mode(SiloMode.CONTROL):
-                    import_in_global_scope(empty_backup_json, printer=NOOP_PRINTER)
-                    # Simulate clearing database if reset_models is called. This is a hacky way
-                    # to get this working for tests, since otherwise db transactions won't work on default db
-                    if reset_models.called:
-                        clear_database()
+                import_fn(empty_backup_json, printer=NOOP_PRINTER)
+
+    @mock.patch("sentry.backup.imports.is_split_db", return_value=False)
+    def test_clears_existing_models_in_global_scope(self, is_split_db):
+        if SiloMode.get_current_mode() != SiloMode.MONOLITH:
+            return
+
+        create_default_projects()
+        self.import_empty_backup_file(import_in_global_scope)
 
         for dependency in dependencies():
             model = get_model(dependency)
-            assert model.objects.count() == 0
-            with connections[router.db_for_read(model)].cursor() as cursor:
-                cursor.execute(f"SELECT MAX(id) FROM {model._meta.db_table}")
+            assert model.objects.count() == 0  # type: ignore
```

Comment:
Nit: odd that this requires a `type: ignore`. Wonder if asserting that `model is not None` would fix this, since the type checker would now know that `model` is an instance of a django model for sure?